packages/dds/map/src/mapIncremental.ts

```diff
@@ -0,0 +1,466 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ISequencedDocumentMessage,
+	MessageType,
+	SummaryType,
+} from "@fluidframework/protocol-definitions";
+import {
+	IChannelAttributes,
+	IFluidDataStoreRuntime,
+	IChannelStorageService,
+	IChannelServices,
+	IChannelFactory,
+} from "@fluidframework/datastore-definitions";
+import {
+	IIncrementalContext,
+	ISummaryTreeWithStats,
+	ITelemetryContext,
+} from "@fluidframework/runtime-definitions";
+import { readAndParse } from "@fluidframework/driver-utils";
+import { IFluidSerializer, SharedObject } from "@fluidframework/shared-object-base";
+import { SummaryTreeBuilder } from "@fluidframework/runtime-utils";
+import { ISharedMap, ISharedMapEvents } from "./interfaces";
+import { IMapDataObjectSerializable, IMapOperation, MapKernel } from "./mapKernel";
+import { pkgVersion } from "./packageVersion";
+
+interface IMapSerializationFormat {
+	blobs?: string[];
+	content: IMapDataObjectSerializable;
+}
+
+const snapshotFileName = "header";
+
+function addBlobOrHandle(
+	parentPath: string,
+	blobName: string,
+	previousBlob: string,
+	currentBlob: string,
+	builder: SummaryTreeBuilder,
+) {
+	if (currentBlob === previousBlob) {
```

Comment:
And I think it's also incorrect. If summary A was successful, and summary B was not, and how we are producing summary C, then DDS will refer to blobs in summary B that does not exists, but storage will assume that's a reference to a blob in summary A. result - it either fails (such blob did not exist in summary A), or incorrect summary (even worse).